docs/whitepaper/whitepaper.md

```diff
@@ -0,0 +1,591 @@
+|   | Authentication Strategies | `Array<string>` | `--authentication-strategies`<br><br>`AUTHENTICATION_STRATEGIES`<br><br>`authenticationStrategies` | Optional. Specifies the fully qualified name, version and exported module of one or more npm packages that are to be loaded and used as the providers for the authentication strategies. For example to use PassportJS's OpenID Connect strategy one with specify the value `["passport-oidc-strategy@0.1.1###Strategy"]` which will get parsed as a JSON string containing an array of strings.                        |
+|   | Authentication Options    | `Array<string>` | `--authentication-options`<br><br>`AUTHENTICATION_OPTIONS`<br><br>`authenticationOptions`          | Used to provide arguments to the constructors (or factory functions) exported by the modules specified by `AUTHENTICATION_STRATEGIES`. For example, in this configuration parameter you can specify the callback URL for an Open ID Connect provider of your choice, the client ID, client secret, etc. Important: The order in which the items appear have to match the order of items in `AUTHENTICATION_STRATEGIES`. |
+|   | Package Registries        | `Array<string>` | `--package-registries`<br><br>`PACKAGE_REGISTRIES`<br><br>`packageRegistries`                      | Optional. Defaults to the public npm registry at `https://registry.npmjs.org/`. Can be used to specify private registries in the event of closed source plugins. If multiple registry URLs are provided they all will be tried in-order at bootstrap time.                                                                                                                                                              |
+
+##### 4.1.1.1.4 Plugin Loading/Validation
+
+Plugin loading happens through NodeJS's built-in module loader and the validation is performed by the Node Package Manager tool (npm) which verifies the byte level integrity of all installed modules.
+
+#### 4.1.1.2 cmd-api-server
+
+A command line application for running the API server that provides a unified REST based HTTP API for calling code.
+
+By design this is stateless and horizontally scalable.
+
+Comes with Swagger API definitions.
+
+#### 4.1.1.3 cmd-validator
+
+Command line application to run a single `BIF` validator node.
+
+#### 4.1.1.4 sdk-javascript
+
+Javascript SDK (bindings) for the RESTful HTTP API provided by `cmd-api-server`.
+Compatible with both NodeJS and Web Browser (HTML 5 DOM + ES6) environments.
+
+#### 4.1.1.5 keychain
```

Comment:
Can you elaborate on what, exactly you are using (or aspire to use) for this?  Thanks!

Reply:
@hartm Currently I am eyeing Vault [1] as the primary (default) candidate. 
Based on the fact that `BAF` already uses it, I made the assumption that the licensing of Vault is not an issue (fingers crossed).

I also foresee a demand for the keychain's backing store being pluggable in case someone wants to use the solutions native to their cloud provider such as AWS SM [2] or KMS [3] and whatever the equivalent of these are on the other big cloud providers.
 
I will add a dedicated section with this info under `Plugin Architecture`. 
Thank you for the feedback, it really helps in figuring out what's missing.

[1] https://www.vaultproject.io/
[2] https://aws.amazon.com/secrets-manager/
[3] https://aws.amazon.com/kms/

Reply:
Expanded on the keychain APIs in a new section of the plugin architecture here:
https://github.com/petermetz/blockchain-integration-framework/commit/826da52613f77ea408efe6ce7a282991cb10f915